### DISCUSSION TITLE
subprocess "systemd --user" cannot shield SIGINT

### DISCUSSION THREAD

**Discussion by iasunsea** [rhinstaller/anaconda#4565] [Q&A]
i test it( https://github.com/rhinstaller/anaconda/commit/d8060d01a01e3d5b187ae4388f10b0d0c2c0c4f3 )，and find with GUI in F1 ctrl+c ， it also  stop with traceback. And then i find i used the old version, and there is another one subprocess, "systemd --user". But i use the same 'preexec_fn' to shield SIGINT, i failed. So i want to know why ?

**Comment by VladimirSlavik** [comment]
Hi! Sorry. Unfortunately, I have no answers without code examples.

**Reply by iasunsea** [reply]
https://github.com/iasunsea/anaconda/commit/3665c20ce4c15ef83b43eb5be00085771cfc6d17, like this.
then the systemd also response the 'ctrl+c' in the GUI F1

**Reply by iasunsea** [reply]
@VladimirSlavik help me to set a strategy
